core/state_transition.go

```diff
@@ -142,16 +154,30 @@ func IntrinsicGas(data []byte, contractCreation, isHomestead bool, isEIP2028 boo
 }
 
 // NewStateTransition initialises and returns a new state transition object.
-func NewStateTransition(evm *vm.EVM, msg Message, gp *GasPool) *StateTransition {
-	return &StateTransition{
-		gp:       gp,
-		evm:      evm,
-		msg:      msg,
-		gasPrice: msg.GasPrice(),
-		value:    msg.Value(),
-		data:     msg.Data(),
-		state:    evm.StateDB,
+func NewStateTransition(evm *vm.EVM, msg Message, gp, gp1559 *GasPool) *StateTransition {
+	isEIP1559 := evm.ChainConfig().IsEIP1559(evm.Context.BlockNumber) && msg.GasPrice() == nil && msg.MaxMinerBribe() != nil && msg.FeeCap() != nil && evm.Context.BaseFee != nil && gp1559 != nil
+	st := &StateTransition{
+		gp:        gp,
+		gp1559:    gp1559,
+		evm:       evm,
+		msg:       msg,
+		gasPrice:  msg.GasPrice(),
+		value:     msg.Value(),
+		data:      msg.Data(),
+		state:     evm.StateDB,
+		isEIP1559: isEIP1559,
 	}
+	if isEIP1559 {
+		// # bribe is capped such that base fee is filled first
+		// bribe_per_gas = min(transaction.max_miner_bribe_per_gas, transaction.fee_cap_per_gas - block.base_fee)
+		// # signer pays both the bribe and the base fee
+		// effective_gas_price = bribe_per_gas + block.base_fee
+		st.eip1559GasPrice = new(big.Int).Add(evm.Context.BaseFee, math.BigMin(
```

Comment:
Why can I not find `math.BigMin` in [math/big](https://golang.org/pkg/math/big/)?

Reply:
It's not in `math/big` but in our `common/math` https://godoc.org/github.com/ethereum/go-ethereum/common/math#BigMin